src/MergeContext.js

```diff
@@ -454,6 +454,252 @@ class BranchPosition
+        this._trailer = null;
+
+        // GitHub-suggested commit author
+        // uses {name, email, date} format
+        assert(defaultAuthor);
+        this._defaultAuthor = defaultAuthor;
+
+        // optional commit author specified by 'Authored-by' header field
+        // uses {name, email} format
+        // overwrites this._defaultAuthor's name and email
+        this._customAuthor = null;
+
+        this._checkRaw(this._title);
+
+        if (rawPr.body !== undefined && rawPr.body !== null)
+            this._parse(rawPr.body);
+    }
+
+    // complete message for the future commit
+    whole() {
+        assert(this._title.length); // the only required part
+        let message = this._title;
+        if (this._body !== null)
+            message += '\n\n' + this._body;
+        if (this._trailer !== null)
+            message += '\n\n' + this._trailer;
+        return message;
+    }
+
+    // the future commit author in the {name, email, date} format
+    author()
+    {
+        if (this._customAuthor === null)
+            return this._defaultAuthor;
+        return {name: this._customAuthor.name, email: this._customAuthor.email, date: this._defaultAuthor.date};
+    }
+
+    // returns the position of the first non-ASCII_printable character (or -1)
+    _invalidCharacterPosition(str) {
+        const prohibitedCharacters = /[^\u{20}-\u{7e}]/u;
+        const match = prohibitedCharacters.exec(str);
+        return match ? match.index : -1;
+    }
+
+    // removes leading empty lines and trims the end
+    _trim(str) {
+        // cannot just use trim() to preserve the first line indentation (if any).
+        return str.replace(/^\s*\n+/g, '').trimEnd();
+    }
+
+    _paragraphLabel(str) {
+        if (/^\s{4}/.test(str)) // not a regular paragraph but a quotation
+            return null;
+        const rawLabel = str.match(/^\s*(\S+)\s*:/);
+        if (!rawLabel)
+            return null;
+        const label = rawLabel[1].toLowerCase();
+        return label[0].toUpperCase() + label.substring(1);
+    }
+
+    _startsWithFieldName(str) {
+        const label = this._paragraphLabel(str);
+        return (label && label.includes('-')) ? label : null;
+    }
+
+    // basic checks for an unparsed message
+    _checkRaw(message) {
+        // CRs in CRLF sequences already removed
+        // other CRs are not treated specially (and are banned)
+        const lines = message.split('\n');
+        for (let i = 0; i < lines.length; ++i) {
+            const line = lines[i];
+            const invalidPosition = this._invalidCharacterPosition(line);
+
+            if (invalidPosition !== -1)
+                throw new Error(`bad character at line ${i}, offset ${invalidPosition}`);
+
+            // allow excessively long whitespace-only lines
+            // that some copy-pasted PR descriptions may include
+            const trimmedLine = this._trim(line);
+            if (trimmedLine.length > 72)
```

Comment:
Ok, but why trailer? So that we could allow, e.g., long user_name+email lines for Co-authered-by - and trailers cannot be multiline?

Reply:
> So that we could allow, e.g., long user_name+email lines

Exactly. For example, the longest existing entry in Squid CONTRIBUTORS is 70 characters long, and that is without the header name...

> and trailers cannot be multiline?

I do not know whether they can be, but our current PR code pretty much assumes that the end of the line is the end of the field.

Reply:
BTW, I failed to add you as the author of PR 1109 because of that limit: The line was 77 characters long.

There may be an even more general use case here, accommodating long URLs and such. This is one of the reasons I would not rush to implement this enhancement in this PR.